Code/StaminaMeter/BaseStaminaMeterDisplay.cs

```diff
@@ -29,45 +51,77 @@ public override void Added(Entity entity)
             level = SceneAs<Level>();
             // hopefully getting player once is enough, it hasn't caused any issues yet
             player = level.Tracker.GetEntity<Player>();
+            playerData = DynamicData.For(player);
             currentStamina = player.Stamina;
+            currentStarFlyTime = playerData.Get<float>("starFlyTimer");
             drawStamina = currentStamina;
         }
 
         public override void Update()
         {
             base.Update();
             currentStamina = player.Stamina;
+            currentStarFlyTime = playerData.Get<float>("starFlyTimer");
             if (!StaminaMeterModule.Settings.ShowMoreThanMaxStamina)
             {
-                currentStamina = Calc.Min(currentStamina, 110f);
+                currentStamina = Calc.Min(currentStamina, maxStamina);
+                currentStarFlyTime = Calc.Min(currentStarFlyTime, maxStarFlyTime);
             }
             drawStamina = Calc.Approach(drawStamina, currentStamina, 250f * Engine.DeltaTime);
-            if (drawStamina != 110f && !player.Dead)
+            if (drawStamina != maxStamina && !player.Dead)
             {
                 displayTimer = 0.75f;
             }
             else if (displayTimer > 0f)
             {
                 displayTimer -= Engine.DeltaTime;
             }
-
-            Visible = (displayTimer > 0f) && !(level.Paused && StaminaMeterModule.Settings.HideWhilePaused);
+            if (player.StateMachine == Player.StStarFly)
+            {
+                Visible = StaminaMeterModule.Settings.StarFlyTimerEnabled && !player.Dead;
+            }
+            else
+            {
+                Visible = StaminaMeterModule.Settings.StaminaMeterEnabled && displayTimer > 0f;
+            }
```

Comment:
The logic here is probably still a little bit weird. Stamina meters should not be displayed after stopping a star-fly because starting a star-fly would refill stamina. The video is recorded from the room named "19" in 6a. In the video, green stamina meters are displayed shortly after dashes and before starting the next star-fly. However, I cannot observe this glitch in the first room of 6a. I don't know why this glitch happens.

https://github.com/SSM240/StaminaMeter/assets/86339734/3352b3cc-97ea-4697-b2d9-523eb4c92f16